### PR TITLE
Upgrade to latest MQTT.js

### PR DESCRIPTION
**Pull request by hardillb**
part of #263
## Description

Update mqtt.js and engines section of package.json.

Also had to change the client.connect()` call slightly to get it to work properly.

## Related Issue(s)

#263

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
These are the changes I needed to make to get #267 to work properly

File: lib/mqtt.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ class MQTTClient {
                 agent: getWSProxyAgent(this.config.brokerURL)
             }
         }
-        this.client = mqtt.connect(brokerURL, this.brokerConfig)
+        this.client = mqtt.connect(brokerURL.href, this.brokerConfig)
```
Comment (Steve-Mcl) [inline lib/mqtt.js:69]: As far as I can see, this will cause the ERR_INVALID_URL error mentioned in the code above.

TLDR;
Using `brokerURL.href` results in sending the original URL (e.g. 'ws://[::1]:8883/'`) which gets manually deconstructed and reconstructed in the bowels of the MQTT lib as 'ws://::1:8883/'` resulting in the original problem.


DETAILS:

REF: https://github.com/mqttjs/MQTT.js/issues/1569

I have looked at the src for V5 and they [still use](https://github.com/mqttjs/MQTT.js/blob/accd78e38aa82c8cc1ea04029e56494276776c87/lib/connect/index.js#L64) the depreciated `url.parse` to deconstruct the URL then reconstruct it manually which is what out workaround was fixing.


The only thing I am not 100% certain of is if the `import url` at the top of the v5 code causes the code to use the old depreciated `url.parse` or the new `new URL` methods after the ts is compiled.  If it is the latter, the problem goes away.



To confirm this I will need to pull this branch and do local testing with IPV6

Comment (Steve-Mcl) [reply lib/mqtt.js:69]: Unfortunately, this does indeed cause an error when a IPV6 IP URL is used.

Looking through the `MQTT.js` src, i cannot see how passing a string vs passing the URL object makes any difference to reconnect options.

If I have missed something there may be an alternative route. If we must send a string URL then there is an opportunity to intercept and update the final connection URL by setting the connect opts function `transformWsUrl` 

NOTE: I have raised a PR against the linked issue on `MQTT.js`

Comment (hardillb) [reply lib/mqtt.js:69]: Passing the object was leading to it trying to  connect to localhost in the device agent even when the URL pointed to staging.

**Comment by Steve-Mcl** [comment]
Looks good ben but [the whole workaround code](https://github.com/FlowFuse/device-agent/pull/306/files#diff-22d60ae947b2cc2a1180672627a138f507749c3f98792237cd599d075ec38998R52-R62) can go in the bin now.

I would add a suggestion to save you messing on but its out of the touched code and GH wont let me.

```
        // PROBLEM: ipv6 ws addresses cannot connect
        // INFO: Calling mqtt.connect('http://[::1]:8883') fails with error  ERR_INVALID_URL
        // INFO: Calling mqtt.connect(new URL('http://[::1]:8883')) fails because `connect` only accepts a `string` or `url.parse` object
        // INFO: Calling mqtt.connect(url.parse('http://[::1]:8883') fails because unlike new URL, url.parse drops the square brackets off hostname
        //       (mqtt.js disassembles and reassembles the url using hostname + port so `ws://[::1]:8883` becomes `ws://::1:8883`)
        // INFO: WS src code uses `new URL` so when `mqtt.js` passes the reassembled IP `http://::1:8883`, it fails with error ERR_INVALID_URL
        // SEE: https://github.com/mqttjs/MQTT.js/issues/1569
        // eslint-disable-next-line n/no-deprecated-api
        const brokerURL = url.parse(this.config.brokerURL)
        const _url = new URL(this.config.brokerURL)
        brokerURL.hostname = _url.hostname
```

simply becomes 

```
        const brokerURL = new URL(this.config.brokerURL)
```

**Review by Steve-Mcl — APPROVED** [review]
Approved...

MQTT JS v5.10 is now released - do we want to bump to `latest` or leave as is here?

Will let you decide to bump or merge :)

**Comment by hardillb** [comment]
Not merging until we decide the release strategy for this, as I think  we need to tag a NodeJS v14 support branch

**Review by knolleary — APPROVED** [review]

**Comment by hardillb** [comment]
Just a moment, rebasing to pick up the package-lock.json
